presto-hive-function-namespace/src/main/java/com/facebook/presto/hive/functions/HiveFunctionNamespaceManager.java

```diff
@@ -0,0 +1,269 @@
+import static com.facebook.presto.hive.functions.scalar.HiveScalarFunction.createHiveScalarFunction;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_USER_ERROR;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class HiveFunctionNamespaceManager
+        implements FunctionNamespaceManager<HiveFunction>
+{
+    public static final String ID = "hive-functions";
+
+    private final ClassLoader classLoader;
+    private final LoadingCache<FunctionKey, HiveFunction> functions;
+    private final HiveFunctionRegistry hiveFunctionRegistry;
+    private final TypeManager typeManager;
+
+    @Inject
+    public HiveFunctionNamespaceManager(
+            @ForHiveFunction ClassLoader classLoader,
+            HiveFunctionRegistry hiveFunctionRegistry,
+            TypeManager typeManager)
+    {
+        this.hiveFunctionRegistry = requireNonNull(hiveFunctionRegistry, "hiveFunctionRegistry is null");
+        this.classLoader = requireNonNull(classLoader, "classLoader is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.functions = CacheBuilder.newBuilder()
+                .build(CacheLoader.from(this::initializeFunction));
+    }
+
+    @Override
+    public void setBlockEncodingSerde(BlockEncodingSerde blockEncodingSerde)
+    {
+        return;
+    }
+
+    public FunctionNamespaceTransactionHandle beginTransaction()
+    {
+        return new EmptyTransactionHandle();
+    }
+
+    public void commit(FunctionNamespaceTransactionHandle transactionHandle)
+    {
+    }
+
+    public void abort(FunctionNamespaceTransactionHandle transactionHandle)
+    {
+    }
+
+    public void createFunction(SqlInvokedFunction function, boolean replace)
+    {
+        throw new PrestoException(GENERIC_USER_ERROR, format("Cannot create function in hive function namespace: %s", function.getSignature().getName()));
+    }
+
+    public void alterFunction(QualifiedObjectName functionName, Optional<List<TypeSignature>> parameterTypes, AlterRoutineCharacteristics alterRoutineCharacteristics)
+    {
+        throw new PrestoException(GENERIC_USER_ERROR, format("Cannot alter function in hive function namespace: %s", functionName));
+    }
+
+    public void dropFunction(QualifiedObjectName functionName, Optional<List<TypeSignature>> parameterTypes, boolean exists)
+    {
+        throw new PrestoException(GENERIC_USER_ERROR, format("Cannot drop function in hive function namespace: %s", functionName));
+    }
+
+    @Override
+    public Collection<HiveFunction> listFunctions(Optional<String> likePattern, Optional<String> escape)
+    {
+        return functions.asMap().values();
```

Comment:
This will only list the functions in cache right? This api is used in `SHOW FUNCTIONS`. If you only list the functions in cache, it's like unless people somehow know abut it, they'd never be able to discover the function. I guess the problem is that for hive functions, we don't know all signatures up front, so it's impossible to list out all functions with function signature. But it would be beneficial to at least list out all function names. Let's sync up offline.